### PR TITLE
Add a simple integration test for Gunicorn worker

### PR DESCRIPTION
**Pull request by messense**

### CODE REVIEW

**Comment by messense** [comment]
Interesting, it hangs when run all the tests but works when run the specific test.

**Comment by seemethere** [comment]
I experienced this as well when writing the original multiprocessing test.

**Comment by messense** [comment]
When I run `gunicorn --bind 127.0.0.1:1337 --worker-class sanic.worker.GunicornWorker examples.simple_server:app` locally it keeps spawn new workers.

After removing the commits in #742 it works again. So I guess that PR breaks Gunicorn worker... cc @r0fls

**Comment by r0fls** [comment]
@messense thanks for catching that, let me fiddle -- you didn't pinpoint the breakage did you? The previous state was incompatible with passing an initialized socket in general, which defeated the purpose of accepting a sock argument. So we should find something that works for both...

**Comment by r0fls** [comment]
@messense actually that still works for me with the start command:

```
gunicorn --bind 127.0.0.1:1337 --worker-class sanic.worker.GunicornWorker examples.simple_server:ap
```

File: sanic/app.py [hunk 1]
```diff
@@ -543,6 +543,9 @@ def run(self, host=None, port=None, debug=False, ssl=None,
         :param protocol: Subclass of asyncio protocol class
         :return: Nothing
         """
+        if sock is None:
+            host, port = host or "127.0.0.1", port or 8000
```
Comment (r0fls) [inline sanic/app.py]: this should be unneeded with https://github.com/channelcat/sanic/pull/742/files

Can you rebase this change against the HEAD?

**Comment by r0fls** [comment]
@messense oh sorry, you said locally it keeps spawning new workers. I didn't check for that. However I checked now and it does seem to be producing new processes for each request before this commit:

curl command:
```
r0fls 00:14 sanic ((HEAD detached at 4963154))$ curl localhost:1337
{"test":true}
r0fls 00:14 sanic ((HEAD detached at 4963154))$ curl localhost:1337
{"test":true}
r0fls 00:14 sanic ((HEAD detached at 4963154))$ curl localhost:1337
{"test":true}
r0fls 00:14 sanic ((HEAD detached at 4963154))$ curl localhost:1337
```

output from gunicorn worker:
```
2017-05-23 00:14:49 - (network)[INFO][127.0.0.1:62876]: GET http://localhost:1337/  200 13
2017-05-23 00:14:51 - (network)[INFO][127.0.0.1:62878]: GET http://localhost:1337/  200 13
2017-05-23 00:14:52 - (network)[INFO][127.0.0.1:62880]: GET http://localhost:1337/  200 13
2017-05-23 00:14:53 - (network)[INFO][127.0.0.1:62882]: GET http://localhost:1337/  200 13
2017-05-23 00:14:53 - (network)[INFO][127.0.0.1:62884]: GET http://localhost:1337/  200 13
```

Note the PIDs are apparently changing: 62876, 62878, 62880. Also `4963154` is before #742.

**Comment by messense** [comment]
@r0fls I don't think that are PIDs. I think it's  request source ip and port.

https://github.com/channelcat/sanic/blob/master/sanic/config.py#L36

File: sanic/app.py (hunk 1 above)
Comment (messense) [inline sanic/app.py]: `Current branch feature/gunicorn-worker-test-case is up to date.`

**Comment by r0fls** [comment]
ah ha, derp :D was about to ask why the PID was logged there.. ok so how are you seeing that it's spawning more workers? 🤔

**Comment by messense** [comment]
```
2017-05-23 15:21:59 - (sanic)[INFO]: Goin' Fast @ http://127.0.0.1:8000
[2017-05-23 15:21:59 +0800] [54624] [INFO] Booting worker with pid: 54624
2017-05-23 15:21:59 - (sanic)[DEBUG]:
                 ▄▄▄▄▄
        ▀▀▀██████▄▄▄       _______________
      ▄▄▄▄▄  █████████▄  /                 \
     ▀▀▀▀█████▌ ▀▐▄ ▀▐█ |   Gotta go fast!  |
   ▀▀█████▄▄ ▀██████▄██ | _________________/
   ▀▄▄▄▄▄  ▀▀█▄▀█════█▀ |/
        ▀▀▀▄  ▀▀███ ▀       ▄▄
     ▄███▀▀██▄████████▄ ▄▀▀▀▀▀▀█▌
   ██▀▄▄▄██▀▄███▀ ▀▀████      ▄██
▄▀▀▀▄██▄▀▀▌████▒▒▒▒▒▒███     ▌▄▄▀
▌    ▐▀████▐███▒▒▒▒▒▐██▌
▀▄▄▄▄▀   ▀▀████▒▒▒▒▄██▀
          ▀▀█████████▀
        ▄▄██▀██████▀█
      ▄██▀     ▀▀▀  █
     ▄█             ▐▌
 ▄▄▄▄█▌              ▀█▄▄▄▄▀▀▄
▌     ▐                ▀▀▄▄▄▀
 ▀▀▄▄▀

2017-05-23 15:21:59 - (sanic)[INFO]: Goin' Fast @ http://127.0.0.1:8000
[2017-05-23 15:21:59 +0800] [54625] [INFO] Booting worker with pid: 54625
2017-05-23 15:21:59 - (sanic)[DEBUG]:
                 ▄▄▄▄▄
        ▀▀▀██████▄▄▄       _______________
      ▄▄▄▄▄  █████████▄  /                 \
     ▀▀▀▀█████▌ ▀▐▄ ▀▐█ |   Gotta go fast!  |
   ▀▀█████▄▄ ▀██████▄██ | _________________/
   ▀▄▄▄▄▄  ▀▀█▄▀█════█▀ |/
        ▀▀▀▄  ▀▀███ ▀       ▄▄
     ▄███▀▀██▄████████▄ ▄▀▀▀▀▀▀█▌
   ██▀▄▄▄██▀▄███▀ ▀▀████      ▄██
▄▀▀▀▄██▄▀▀▌████▒▒▒▒▒▒███     ▌▄▄▀
▌    ▐▀████▐███▒▒▒▒▒▐██▌
▀▄▄▄▄▀   ▀▀████▒▒▒▒▄██▀
          ▀▀█████████▀
        ▄▄██▀██████▀█
      ▄██▀     ▀▀▀  █
     ▄█             ▐▌
 ▄▄▄▄█▌              ▀█▄▄▄▄▀▀▄
▌     ▐                ▀▀▄▄▄▀
 ▀▀▄▄▀

2017-05-23 15:21:59 - (sanic)[INFO]: Goin' Fast @ http://127.0.0.1:8000
^C[2017-05-23 15:21:59 +0800] [54597] [INFO] Handling signal: int
[2017-05-23 15:21:59 +0800] [54597] [INFO] Shutting down: Master
```

Keep going until I hit Ctrl-C

**Comment by messense** [comment]
Notice it says `Goin' Fast @ http://127.0.0.1:8000` while I am running

`gunicorn --bind 127.0.0.1:1337 --worker-class sanic.worker.GunicornWorker examples.simple_server:app`

**Comment by messense** [comment]
Make sure you run a `pip install -e .` or `python setup.py develop` before run it.

**Comment by r0fls** [comment]
i didn't realize that was needed for testing with gunicorn 👍  you are correct sir, that is not cool...

**Pull request merged**